shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm

```diff
@@ -750,17 +750,34 @@ static bool ClipRRectContainsPlatformViewBoundingRect(const SkRRect& clip_rrect,
   UIView* flutter_view = flutter_view_.get();
   // Clear the `active_composition_order_`, which will be populated down below.
   active_composition_order_.clear();
+  NSMutableArray* desired_platform_subviews = [NSMutableArray array];
   for (size_t i = 0; i < composition_order_.size(); i++) {
     int64_t platform_view_id = composition_order_[i];
     std::vector<std::shared_ptr<FlutterPlatformViewLayer>> layers = layer_map[platform_view_id];
     UIView* platform_view_root = root_views_[platform_view_id].get();
-    // `addSubview` will automatically reorder subview if it is already added.
-    [flutter_view addSubview:platform_view_root];
+    [desired_platform_subviews addObject:platform_view_root];
     for (const std::shared_ptr<FlutterPlatformViewLayer>& layer : layers) {
-      [flutter_view addSubview:layer->overlay_view_wrapper];
+      [desired_platform_subviews addObject:layer->overlay_view_wrapper];
     }
     active_composition_order_.push_back(platform_view_id);
   }
+
+  NSSet* desired_platform_subviews_set = [NSSet setWithArray:desired_platform_subviews];
+  NSArray* existing_platform_subviews = [flutter_view.subviews
+      filteredArrayUsingPredicate:[NSPredicate predicateWithBlock:^BOOL(id object,
+                                                                        NSDictionary* bindings) {
+        return [desired_platform_subviews_set containsObject:object];
```

Comment:
`BringLayersIntoView` is meant to be called after `RemoveUnusedLayers();`, so I think any view in `existing_platform_subviews` should already be in `desired_platform_subviews_set`? Did you actually see this filter reduces the size of `existing_platform_subviews`?

Maybe instead we can NSAssert it?

Reply:
> Did you actually see this filter reduces the size of existing_platform_subviews?

Yes, and my understanding is that there are non-platform views added somewhere

Reply:
Not sure if it's dummy scroll view on the top status bar that I saw (which enables tap-to-scroll for table views), but that's not important.

Reply:
ah ok forgot about those. Yeah makes sense.